src/scripts/create_norglm_multisum.py

```diff
@@ -0,0 +1,71 @@
+"""Create the NorGLM NO-multi summarisation dataset."""
+
+import pandas as pd
+from constants import MAX_NUM_CHARS_IN_ARTICLE, MIN_NUM_CHARS_IN_ARTICLE
+from datasets import Dataset, DatasetDict, Split, load_dataset
+from huggingface_hub import HfApi
+from requests import HTTPError
+
+
+def main():
+    """Create the NorGLM NO-multi summarisation dataset and upload to HF Hub."""
+    dataset_id = "NorGLM/NO-Multi-QA-Sum"
+
+    dataset = load_dataset(dataset_id, split="train", token=True)
+    assert isinstance(dataset, Dataset)
+
+    dataset = dataset.rename_columns(
+        column_mapping=dict(article="text", summary="target_text")
+    )
+
+    df = dataset.to_pandas()
+    assert isinstance(df, pd.DataFrame)
+
+    # Only work with samples where the text is not very large or small
+
+    lengths = df.text.str.len()
+    lower_bound = MIN_NUM_CHARS_IN_ARTICLE
+    upper_bound = MAX_NUM_CHARS_IN_ARTICLE
+    df = df[lengths.between(lower_bound, upper_bound)]
+
+    # Create validation split
+    val_size = 64
+    val_df = df.sample(n=val_size, random_state=4242)
+
+    # Create test split
+    test_size = 256
+    filtered_df = df[~df.index.isin(val_df.index)]
+    test_df = filtered_df.sample(n=test_size, random_state=4242)
+
+    # Create train split
+    train_size = 147
+    filtered_df = filtered_df[~filtered_df.index.isin(test_df.index)]
+    train_df = filtered_df.sample(n=train_size, random_state=4242)
```

Comment:
No need to specify the train size, as it is simply the rest:

```suggestion
    train_df = filtered_df[~filtered_df.index.isin(test_df.index)]
```